resources/qml/PrinterOutput/ManualPrinterControl.qml

```diff
@@ -429,6 +428,120 @@ Item
             }
         }
 
+        Row
+        {
+            id: customCommandInputRow
+
+            width: base.width - 2 * UM.Theme.getSize("default_margin").width
+            height: childrenRect.height + UM.Theme.getSize("default_margin").width
+            anchors.left: parent.left
+            anchors.leftMargin: UM.Theme.getSize("default_margin").width
+
+            spacing: UM.Theme.getSize("default_margin").width
+
+            Label
+            {
+                text: catalog.i18nc("@label", "Send G-code")
+                color: UM.Theme.getColor("setting_control_text")
+                font: UM.Theme.getFont("default")
+
+                width: Math.floor(parent.width * 0.4) - UM.Theme.getSize("default_margin").width
+                height: UM.Theme.getSize("setting_control").height
+                verticalAlignment: Text.AlignVCenter
+            }
+
+            Row
+            {
+                // Input field for custom G-code commands.
+                Rectangle
+                {
+                    id: customCommandControl
+
+                    // state
+                    visible: printerModel != null ? printerModel.canPreHeatBed: true
```

Comment:
canPreHeatBed has nothing to do with this functionality. canControlManually would be more appropriate, though a separate "ability" might be introduced (canSendRawGcode ?)

Reply:
Added this.